tests/foreman/cli/test_ansible.py

```diff
@@ -561,3 +566,112 @@ def test_positive_install_ansible_collection(
         assert result['success'] == '1'
         collection_path = client.execute('ls ~/ansible_collections').stdout
         assert 'oasis_roles' in collection_path
+
+    @pytest.mark.tier3
+    @pytest.mark.no_containers
+    @pytest.mark.rhel_ver_list([settings.content_host.default_rhel_version])
+    @pytest.mark.parametrize('auth_type', ['admin', 'non-admin'])
+    def test_positive_ansible_variables_imported_with_roles(
+        self, request, auth_type, target_sat, module_org, module_ak_with_cv, rhel_contenthost
+    ):
+        """Verify that when Ansible roles are imported, their variables are imported simultaneously
+
+        :id: 107c53e8-5a8a-4291-bbde-fbd66a0bb85e
+
+        :steps:
+            1. Register a content host with satellite
+            2. Create a custom role with variables and import it into satellite
+            3. Assert that the role is imported along with associated variables
+            4. Assign that role to a host and verify the role is assigned to the host
+            5. Run the Ansible role
+
+        :expectedresults: Verify variables associated to role are also imported along with roles
+
+        :verifies: SAT-28198
+        """
+        username = settings.server.admin_username
+        password = settings.server.admin_password
+        if auth_type == 'non-admin':
+            username = gen_string('alpha')
+            user = target_sat.cli_factory.user(
+                {
+                    'admin': False,
+                    'login': username,
+                    'password': password,
+                    'organization-ids': module_org.id,
+                }
+            )
+            target_sat.cli.User.add_role(
+                {'id': user['id'], 'login': username, 'role': 'Ansible Roles Manager'}
+            )
+
+        @request.addfinalizer
+        def _finalize():
+            result = target_sat.cli.Ansible.roles_delete({'name': SELECTED_ROLE})
+            assert f'Ansible role [{SELECTED_ROLE}] was deleted.' in result[0]['message']
+            target_sat.execute(f'rm -rvf /etc/ansible/roles/{SELECTED_ROLE}')
+
+        SELECTED_ROLE = gen_string('alphanumeric')
+        playbook = f'{robottelo_tmp_dir}/playbook.yml'
+        vars = f'{robottelo_tmp_dir}/vars.yml'
+        target_sat.execute(f'ansible-galaxy init --init-path /etc/ansible/roles/ {SELECTED_ROLE}')
+        tasks_file = f'/etc/ansible/roles/{SELECTED_ROLE}/tasks/main.yml'
+        vars_file = f'/etc/ansible/roles/{SELECTED_ROLE}/{"defaults" if is_open("SAT-28198") else "vars"}/main.yml'
```

Comment:
When`SAT-28198` is fixed, it will only check /vars path for ansible variables and not both `defaults` and `vars` ?

Reply:
yes, correct, vars is something we should check, and if required we can cover defaults as well in future, but it'll require changing the playbook as there's only one place where we can use ansible variable

Reply:
Ah! Maybe I need more info on default and vars. @nofaralfasi I would like to know your opinion before I resolve this conversation.

Reply:
@shweta83 I'd suggest to check the thread above

Reply:
Based on the discussion with @Gauravtalreja1, I am good with the changes.